wordpress-mobile/WordPress-Android
# Jetpack Section: Backup Screen - Rename Backup Feature Flags

Parent #13629

This PR mainly renames the `Backup` related feature flags. However, as part of this PR additional **EXTRA** changes were also made, see below:
- `Backup` feature flag was changed to `BackupDownload` @zwarm 
- `Backups` feature flag was changed to `BackupScreen` @ParaskP7 
- **EXTRA** - `Scan` feature flag was change to `ScanScreen` @ashiagr 
- **EXTRA** - All above and the `Restore` feature flag were moved to the appropriate `config` package.

To test:
- Launch app and go to `My Site` tab.
- Go to `My Site` -> `Toolbar Avatar` -> `App Settings` -> `Test feature configuration`.
- Verify all the related feature flags are updated according to the above description and below image.
- Enable/Disable all the related feature flags and make sure they works as expected.

<img src="https://user-images.githubusercontent.com/9729923/104730010-fcc21280-5741-11eb-9d2a-262adf821785.png" width="250" height="500">

PR submission checklist:
